src/main/java/org/truffleruby/interop/InteropNodes.java

```diff
@@ -902,4 +919,105 @@ public Object metaObject(Object value) {
+
+        @Specialization
+        public boolean readableBit(int bits) {
+            return KeyInfo.isExisting(bits);
+        }
+
+    }
+
+    @CoreMethod(names = "readable_bit?", isModuleFunction = true, required = 1)
+    public abstract static class HasReadableBitNode extends CoreMethodArrayArgumentsNode {
+
+        @Specialization
+        public boolean readableBit(int bits) {
+            return KeyInfo.isReadable(bits);
+        }
+
+    }
+
+    @CoreMethod(names = "writable_bit?", isModuleFunction = true, required = 1)
+    public abstract static class HasWritableBitNode extends CoreMethodArrayArgumentsNode {
+
+        @Specialization
+        public boolean writableBit(int bits) {
+            return KeyInfo.isWritable(bits);
+        }
+
+    }
+
+    @CoreMethod(names = "invocable_bit?", isModuleFunction = true, required = 1)
+    public abstract static class HasInvocableBitNode extends CoreMethodArrayArgumentsNode {
+
+        @Specialization
+        public boolean invocableBit(int bits) {
+            return KeyInfo.isInvocable(bits);
+        }
+
+    }
+
+    @CoreMethod(names = "internal_bit?", isModuleFunction = true, required = 1)
+    public abstract static class HasInternalBitNode extends CoreMethodArrayArgumentsNode {
+
+        @Specialization
+        public boolean internalBit(int bits) {
+            return KeyInfo.isInternal(bits);
+        }
+
+    }
+
+    @CoreMethod(names = "existing_bit", isModuleFunction = true)
+    public abstract static class ExistingBitNode extends CoreMethodArrayArgumentsNode {
+
+        @Specialization
+        public int existingBit() {
+            // KeyInfo builders are existing by default
+            return KeyInfo.newBuilder().build();
+        }
+
+    }
+
+    @CoreMethod(names = "readable_bit", isModuleFunction = true)
+    public abstract static class ReadableBitNode extends CoreMethodArrayArgumentsNode {
+
+        @Specialization
+        public int readableBit() {
+            return KeyInfo.newBuilder().setReadable(true).build();
+        }
+
+    }
+
+    @CoreMethod(names = "writable_bit", isModuleFunction = true)
+    public abstract static class WritableBitNode extends CoreMethodArrayArgumentsNode {
+
+        @Specialization
+        public int writableBit() {
+            return KeyInfo.newBuilder().setWritable(true).build();
+        }
+
+    }
+
+    @CoreMethod(names = "invocable_bit", isModuleFunction = true)
+    public abstract static class InvocableBitNode extends CoreMethodArrayArgumentsNode {
+
+        @Specialization
+        public int invocableBit() {
+            return KeyInfo.newBuilder().setInvocable(true).build();
+        }
+
+    }
+
+    @CoreMethod(names = "internal_bit", isModuleFunction = true)
+    public abstract static class InternalBitNode extends CoreMethodArrayArgumentsNode {
+
+        @Specialization
+        public int internalBit() {
+            return KeyInfo.newBuilder().setInternal(true).build();
+        }
```

Comment:
That's a lot of methods and it's not immediately clear they are composable, how about a single method `key_info_bits(readable, writable, ...)` taking 5 arguments and using only one builder?

Reply:
Done.